extensions/panache/hibernate-reactive-panache/deployment/src/test/java/io/quarkus/hibernate/reactive/panache/test/DuplicateIdEntityTest.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.hibernate.reactive.panache.test;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.builder.BuildException;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class DuplicateIdEntityTest {
+
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .setExpectedException(BuildException.class)
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(DuplicateIdEntity.class));
+
+    @Test
+    void shouldThrow() {
+        fail("A BuildException should have been thrown due to duplicate entity ID");
```

Comment:
So Panache can't deal with composite IDs ?

Reply:
IIRC this detects the case where you have multiple `@Id` annotations but no matching `@IdClass` which is always a blunder.